src/convert/convertContext/recompositionFinalizer.ts

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
this is what needs the most review attention.  It's quite different from the predecessor in the old ConvertContext pile